### DISCUSSION TITLE
How to use external yt-dlp

### DISCUSSION THREAD

**Discussion by clarjon1** [smplayer-dev/smplayer#1032] [Q&A] [answered]
Hello!

I'm trying to use smplayer with windows 10, and I've been coming from using mpv and yt-dlp, both of which are already in my system $PATH variable

As I dont want to be syncing multiple configuration files around my system, I'd really like to be able to use the mpv and yt-dlp on my system! Mostly because my system yt-dlp configuration has been configured for being able to use cookies from a specific browser install so it can access media locked behind login screens!

I've been able to get it to use the mpv install fine from c:/tools/mpv/mpv.exe but I dont see a way to pass a way to make smplayer not pass ` --ytdl --script-opts=ytdl_hook-ytdl_path=C:/Users/clarjon1/.smplayer/yt-dlp` to the binary! This is very frustrating to me, as the yt-dlp I want is in c:/tools/yt-dlp/yt-dlp.exe

Is there an option I'm overlooking or should I put in a feature request to be able to configure this tool that relies on external tools to use a specific install of the external tool?

**Comment by smplayer-dev** [comment, marked as answer]
If you set "support for video sites" to "disabled" in preferences -> network, smplayer won't pass any --yt-dlp options. You can then pass your own options in preferences -> advanced -> mplayer/mpv -> options.

**Comment by clarjon1** [comment]
cool, that's mostly accurate and does work.

There is a ytdl option that gets passed per logs (`--ytdl=no`) but luckily
passing `--ytdl=yes` overrides that

still would be nice if the option where you can tell it to use ytdl would
let you input the path to the ytdlp install like it does with the mpv
binary but hey at least this works for now.

[image: image.png]

**Comment by raszpl** [comment]
There does exist inside smplayer.ini:
```
[streaming]
streaming\youtube\ytdl_bin=
```
~~but its ignored and does nothing for me~~EDIT: it works great, I didnt notice smplayer.ini moved to its own options subdirectory and was editing wrong file :) warning \ in the path need to be escaped so use `\\`

**Reply by x4740N** [reply]
"path//path"

Works

"path\\\path" does not work

for example `//home//USERNAME//.local//bin//yt-dlp` works but `\\home\\USERNAME\\.local\\bin\\yt-dlp` does not work

**Comment by x4740N** [comment]
Just in case the above comment is deleted because i don't know if github deletes replies to a comment on an issue:

> There does exist inside smplayer.ini:
> 
> ```
> [streaming]
> streaming\youtube\ytdl_bin=
> ```
> 
> ~but its ignored and does nothing for me~EDIT: it works great, I didnt notice smplayer.ini moved to its own options subdirectory and was editing wrong file :) warning \ in the path need to be escaped so use `\\`

"path//path"

Works

"path\\\path" does not work

for example `//home//USERNAME//.local//bin//yt-dlp` works but `\\home\\USERNAME\\.local\\bin\\yt-dlp` does not work
